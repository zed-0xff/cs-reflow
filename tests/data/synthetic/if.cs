void as_is()
{
    if (a)
    {
        foo();
    }
    else
    {
        bar();
    }
}

void keep_true()
{
    var a = true;
    if (a)
    {
        foo();
    }
    else
    {
        bar();
    }
}

void keep_false()
{
    var a = false;
    if (a)
    {
        foo();
    }
    else
    {
        bar();
    }
}

void as_is_indirect()
{
    bool b;
    if (a)
    {
        b = true;
    }
    else
    {
        b = false;
    }

    if (b)
    {
        foo();
    }
    else
    {
        bar();
    }
}

void keep_true_indirect()
{
    var a = true;
    var b;
    
    if (a)
    {
        b = true;
    }
    else
    {
        b = false;
    }

    if (b)
    {
        foo();
    }
    else
    {
        bar();
    }
}

void keep_false_indirect()
{
    var a = false;
    var b;

    if (a)
    {
        b = true;
    }
    else
    {
        b = false;
    }

    if (b)
    {
        foo();
    }
    else
    {
        bar();
    }
}
