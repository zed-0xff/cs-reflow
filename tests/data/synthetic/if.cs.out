void as_is()
{
    if (a)
    {
        foo();
    }
    else
    {
        bar();
    }
}

void keep_true()
{
    foo();
}

void keep_false()
{
    bar();
}

void as_is_indirect()
{
    if (a)
    {
        foo();
    }
    else
    {
        bar();
    }
}

void keep_true_indirect()
{
    foo();
}

void keep_false_indirect()
{
    bar();
}

